File: server/test/sc/server/plugins/TestPlugin.java
package sc.server.plugins;


import sc.api.plugins.IGameInstance;
import sc.api.plugins.IGamePlugin;
import sc.helpers.XStreamKt;
import sc.plugins.PluginDescriptor;
import sc.protocol.helpers.LobbyProtocol;
import sc.shared.ScoreDefinition;

import java.util.Arrays;

@PluginDescriptor(author = "Chuck", name = "TestPlugin", uuid = TestPlugin.TEST_PLUGIN_UUID)
public class TestPlugin implements IGamePlugin {
  public static final String TEST_PLUGIN_UUID = "012345-norris";

  public static final ScoreDefinition SCORE_DEFINITION = new ScoreDefinition("winner");

  @Override
  public IGameInstance createGame() {
    return new TestGame();
  }

  @Override
  public void initialize() {
    LobbyProtocol.registerAdditionalMessages(XStreamKt.getXStream(), Arrays.asList(TestTurnRequest.class, TestGameState.class, TestPlayer.class, TestMove.class));
  }

  @Override
  public void unload() {
  }

  @Override
  public ScoreDefinition getScoreDefinition() {
    return SCORE_DEFINITION;
  }

}

File: server/test/sc/server/plugins/TestPlugin.kt
package sc.server.plugins

import sc.api.plugins.IGameInstance
import sc.api.plugins.IGamePlugin
import sc.api.plugins.IGameState
import sc.plugins.PluginDescriptor
import sc.shared.ScoreDefinition
import java.io.File

@PluginDescriptor(name = "TestPlugin", uuid = TestPlugin.TEST_PLUGIN_UUID)
class TestPlugin: IGamePlugin {
    companion object {
        const val TEST_PLUGIN_UUID = "012345-norris"
    }
    
    override val id: String = TEST_PLUGIN_UUID
    
    override val scoreDefinition: ScoreDefinition =
            ScoreDefinition("winner")
    
    override val gameTimeout = 1000
    
    override fun createGame(): IGameInstance =
            TestGame()
    
    override fun createGameFromState(state: IGameState): IGameInstance =
            TestGame(state as TestGameState)
    
}
